### PR TITLE
Add eslint-plugin-jsx-a11y

### PR DESCRIPTION
**Pull request by zautumnz**
Would take care of #1512 

* [x] Add plugin
* [x] Add script (`yarn run test:lint`)
  * `yarn run test` now runs both mocha and eslint -- to run just mocha, `yarn run test:mocha`
* [x] Fix existing lint errors
* [x] Fix new lint errors (from adding the plugin)

The rules aren't too strict right now, but more could be added ([full rules list](https://github.com/evcohen/eslint-plugin-jsx-a11y/tree/master/docs/rules)).

### CODE REVIEW

**Review by Gargron — APPROVED** [review]
Looks good at first glance, really hope it doesn't subtly break something visually... Are you sure that all these changes do not affect how the elements look?

File: app/assets/javascripts/components/components/load_more.jsx [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import { FormattedMessage } from 'react-intl';
 
 const LoadMore = ({ onClick }) => (
-  <a href='#' className='load-more' onClick={onClick}>
+  <span tabIndex='0' className='load-more' role='button' onClick={onClick}>
```
Comment (zacanger) [inline app/assets/javascripts/components/components/load_more.jsx]: I'm not sure about this one, I haven't run it since making this change. I can run this evening and double check.

Comment (jessebeach) [inline app/assets/javascripts/components/components/load_more.jsx]: You can just put `role="button"` on the `<a href>`. It's better than turning it into a span.

We're working on some big updates to `eslint-plugin-jsx-a11y` that should make it more precise and unambiguous: https://github.com/evcohen/eslint-plugin-jsx-a11y/pull/194

File: app/assets/javascripts/components/components/media_gallery.jsx [hunk 2]
```diff
@@ -160,6 +160,8 @@ const Item = React.createClass({
     } else if (attachment.get('type') === 'gifv') {
       thumbnail = (
         <video
+          tabIndex='0'
+          role='button'
```
Comment (jessebeach) [inline app/assets/javascripts/components/components/media_gallery.jsx]: This is a false-positive of the linter. Fixed in https://github.com/evcohen/eslint-plugin-jsx-a11y/pull/194

We don't want to turn videos into buttons :) It doesn't need the tabIndex or the role attributes.

File: app/assets/javascripts/components/components/load_more.jsx (hunk 1 above)
Comment (zacanger) [inline app/assets/javascripts/components/components/load_more.jsx]: Ah good to know, thank you!

File: app/assets/javascripts/components/features/compose/components/search.jsx [hunk 3]
```diff
@@ -58,7 +58,7 @@ const Search = React.createClass({
 
         <div className='search__icon'>
           <i className={`fa fa-search ${hasValue ? '' : 'active'}`} />
-          <i className={`fa fa-times-circle ${hasValue ? 'active' : ''}`} onClick={this.handleClear} />
+          <i role='button' tabIndex='0' className={`fa fa-times-circle ${hasValue ? 'active' : ''}`} onClick={this.handleClear} />
```
Comment (jessebeach) [inline app/assets/javascripts/components/features/compose/components/search.jsx]: The role, tabIndex and onClick should really be on the div. The icon is content in this case.

And this element needs a label. Use `aria-label`.

**Review by jessebeach — COMMENTED** [review]
Just a couple minor nits.

File: Dockerfile [hunk 4]
```diff
@@ -28,7 +28,7 @@ RUN BUILD_DEPS=" \
     imagemagick \
  && npm install -g npm@3 && npm install -g yarn \
  && bundle install --deployment --without test development \
- && yarn \
+ && yarn --ignore-optional \
```
Comment (zacanger) [inline Dockerfile]: This change is unrelated. Hides all the noise from npm/yarn about `fsevents`.

**Review by Gargron — APPROVED** [review]

**Comment by zacanger** [comment]
Thank you!
